core/src/main/java/io/aiven/klaw/service/TopicSyncControllerService.java

```diff
@@ -595,8 +595,11 @@ private void approveSyncBackTopics(
         }
       } else {
         logUpdateSyncBackTopics.add("Topic created " + topicFound.getTopicname());
-        if (!Objects.equals(syncBackTopics.getSourceEnv(), syncBackTopics.getTargetEnv()))
+        if (!Objects.equals(syncBackTopics.getSourceEnv(), syncBackTopics.getTargetEnv())) {
           createAndApproveTopicRequest(syncBackTopics, topicFound, tenantId);
+          topicFound.setEnvironment(syncBackTopics.getTargetEnv());
+          manageDatabase.addTopicToCache(tenantId, topicFound);
```

Comment:
why would this be required, as this sync is from klaw to cluster. and topic must already be in cache ?

Reply:
The topic is in the cache, however it is assigned only to the original environment.
So if you sync from the Dev env -> Prod env or TST env then until you retstart Klaw it does not show on the Topics page.